src/moin/storage/middleware/indexing.py

```diff
@@ -1144,6 +1145,14 @@ def store_revision(self, meta, data, overwrite=False,
         self.indexer.index_revision(meta, content, backend_name)
         if not overwrite:
             self._current = self.indexer._document(revid=revid)
+            if self._current is None:
+                # under heavy loads, whoosh async writer may be delayed in writing indexes to storage
+                until = time.time() + 20.0
+                while time.time() < until and self._current is None:
+                    time.sleep(2)
+                    self._current = self.indexer._document(revid=revid)
+                if self._current is None:
+                    raise KeyError(revid + '- server overload or corrupt index')
```

Comment:
guess the need for this comes from doing it async now.

assuming that we want this (didn't look into the details), the code could be simpified like (replacing code including 1147):
```
until = time.time() + 20.0
while True:
    self._current = self.indexer._document(revid=revid)
    if self._current is not None:
        break
    # under heavy loads, whoosh async writer may be delayed in writing indexes to storage
    time.sleep(2)
    if time.time() > until:
        raise KeyError(revid + ' - server overload or corrupt index')
```